app/database/operator/server_data_operator/transformers/channel.ts

```diff
@@ -37,9 +40,25 @@ export const transformChannelRecord = ({action, database, value}: TransformerArg
         channel.creatorId = raw.creator_id;
         channel.deleteAt = raw.delete_at;
 
-        // for DM channels do not override the display name
+        // for DM & GM's  channels do not override the display name
         // until we get the new info if there is any
-        channel.displayName = raw.display_name || record?.displayName || '';
+        let displayName;
+        if (raw.type === General.DM_CHANNEL && record?.displayName) {
+            displayName = raw.display_name || record?.displayName;
+        } else if (raw.type === General.GM_CHANNEL) {
+            const rawMembers = raw.display_name.split(',').length;
+            const recordMembers = record?.displayName.split(',').length || rawMembers;
+
+            if (recordMembers < rawMembers) {
```

Comment:
I don't fully understand this part.  If we have a GM with 3 members then, `recordMembers` length will be 3.  

Now, we add a fourth user to this GM, the `rawMembers` will be 4.

The display name on the web app does update. 

So I guess, we should be using the `raw.display_name` if the `rawMembers` length is greater?

Reply:
Alright, let me explain.

Group Message channels or GM's are direct messages with a group of people. When the channel is created it assigns the `display_name` to be `username1, username2, username3, me, username4`  in this example you know that the group message channel has 5 members in total, you and four others.

Then when you show the display_name in the UI regardless of the teammate display name setting, you always remove yourself, so the display_name to be shown in the UI in this case would be `username1, username2, username3, username4` see how `me` has been excluded from it.

* Why don't we use `name` for GM's? cause the `name` of the channel for GM's is actually a in the form of an `id`
* But what happens if I add new members? you cannot add new members to a GM, doing so creates a new GM channel with the current members and the newly added.
<img width="585" alt="image" src="https://user-images.githubusercontent.com/6757047/156544990-54361aec-9dee-4ef1-b8ba-e81ed5c33c2f.png">
* Ok so what happens if I remove members? members cannot be removed from a GM channel.

Hopefully you have enough context here to understand the code